packages/go_router/lib/src/delegate.dart

```diff
@@ -146,11 +147,14 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     // a microtask in case the onExit callback want to launch dialog or other
     // navigator operations.
     scheduleMicrotask(() async {
-      final bool onExitResult = await routeBase.onExit!(
```

Comment:
this won't be nullable otherwise it would return in line 142

Reply:
fixed